src/index.ts

```diff
@@ -45,11 +47,15 @@ app.use((req, res, next) => {
 });
 
 const provider = getProvider();
-const { ovalConfigs } = env;
+const { ovalConfigs, ovalConfigsShared } = env;
 
 // Initialize unlocker wallets for each Oval instance.
-const keyManager = WalletManager.getInstance();
-keyManager.initialize(ovalConfigs);
+const walletManager = WalletManager.getInstance();
+walletManager.initialize(provider, ovalConfigs, ovalConfigsShared);
+
+// Initialize Oval discovery
+const ovalDiscovery = OvalDiscovery.getInstance();
+ovalDiscovery.initialize(provider);
```

Comment:
do we have any idea on how much of a performance impact this has (and all the underlying async calls this triggers) on startup times? given this is running in cloud run, is this problematic?

Reply:
I did some tests, and querying 2 years of blocks in mainnet to find factories takes less than 10 seconds, querying 1 year takes 5.1 seconds. Since this only runs once when we restart the instance or when auto-scaling provisions a new instance, I think it’s fine. Do you agree?